### PR TITLE
feat: ZkSync era contracts

### PR DESCRIPTION
**Pull request by nicholaspai**
Updated ZkSync era contracts that reflect latest on mainnet and testnet

Fixes ACX-1292

### CODE REVIEW

File: contracts/ZkSync_SpokePool.sol [hunk 1]
```diff
@@ -3,107 +3,175 @@ pragma solidity ^0.8.0;
...
+        uint256 originChainId,
+        int64 realizedLpFeePct,
+        int64 relayerFeePct,
+        uint32 depositId,
+        uint32 rootBundleId,
+        bytes memory message,
+        int256 payoutAdjustment,
+        bytes32[] memory proof
+    ) public override(SpokePool) nonReentrant {
+        if (destinationToken == address(wrappedNativeToken)) _depositEthToWeth();
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply contracts/ZkSync_SpokePool.sol:102]: yeah I tested this and added a comment [here](https://github.com/across-protocol/contracts-v2/blob/d0f12635e17a8a10ca33495b205d58149a7ed499/contracts/ZkSync_SpokePool.sol#L141)

File: contracts/ZkSync_SpokePool.sol [hunk 2]
```diff
@@ -3,107 +3,175 @@ pragma solidity ^0.8.0;
...
+    }
+
+    function _bridgeTokensToHubPool(RelayerRefundLeaf memory relayerRefundLeaf) internal override {
+        // SpokePool is expected to receive ETH from the L1 HubPool and currently, withdrawing ETH directly
+        // over the ERC20 Bridge is blocked at the contract level. Therefore, we need to unwrap it before withdrawing.
         emit ZkSyncTokensBridged(relayerRefundLeaf.l2TokenAddress, hubPool, relayerRefundLeaf.amountToReturn);
+        if (relayerRefundLeaf.l2TokenAddress == address(wrappedNativeToken)) {
+            WETH9Interface(relayerRefundLeaf.l2TokenAddress).withdraw(relayerRefundLeaf.amountToReturn); // Unwrap into ETH.
+            // To withdraw tokens, we actually call 'withdraw' on the L2 eth token itself.
+            IL2ETH(l2Eth).withdraw{ value: relayerRefundLeaf.amountToReturn }(hubPool);
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply contracts/ZkSync_SpokePool.sol:157]: We can't unfortunately, WETH is specifically blocked at the contract level which I tested on prod: https://github.com/across-protocol/contracts-v2/blob/d0f12635e17a8a10ca33495b205d58149a7ed499/contracts/ZkSync_SpokePool.sol#L152

File: contracts/ZkSync_SpokePool.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrice32) [reply contracts/ZkSync_SpokePool.sol:102]: Gotcha, makes sense!

File: contracts/ZkSync_SpokePool.sol [hunk 3]
```diff
@@ -3,107 +3,175 @@ pragma solidity ^0.8.0;
...
         zkErc20Bridge = _zkErc20Bridge;
-        zkEthBridge = _zkEthBridge;
-        emit SetZkBridges(address(_zkErc20Bridge), address(_zkEthBridge));
+        emit SetZkBridge(address(_zkErc20Bridge));
+    }
+
+    // L1 addresses are transformed during l1->l2 calls.
+    // See https://github.com/matter-labs/era-contracts/blob/main/docs/Overview.md#mailboxfacet for more information.
+    // Another source: https://github.com/matter-labs/era-contracts/blob/41c25aa16d182f757c3fed1463c78a81896f65e6/ethereum/contracts/vendor/AddressAliasHelper.sol#L28
+    function _applyL1ToL2Alias(address l1Address) internal pure returns (address l2Address) {
```
[the remark this answers is not in the document]
Comment (mrice32) [reply contracts/ZkSync_SpokePool.sol:171]: Makes sense.  SGTM!

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**